components/form.tsx

```diff
@@ -2,14 +2,90 @@
 
 import Link from 'next/link';
 import Image from 'next/image';
+import { useForm } from 'react-hook-form';
+import type { SubmitHandler } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
 
-import { Input, Button } from '@/components';
+import { Input, Button, Spinner, Success } from '@/components';
+
+interface FormFields {
+  email: string;
+  username: string;
+  password: string;
+  confirmPassword: string;
+}
+
+export const UserSchema = z
+  .object({
+    email: z
+      .string()
+      .min(1, 'Email is required.')
+      .email('Please enter a valid email address.')
+      .regex(
+        /^[a-zA-Z0-9._%+-]+@(gmail\.com|hotmail\.com|lightit\.io)$/,
+        'Email must be from one of the allowed domains: gmail.com, hotmail.com, or lightit.io.',
+      )
+      .trim(),
+    username: z
+      .string({ required_error: 'Username is required.' })
+      .min(1, 'Username is required.')
+      .toLowerCase()
+      .regex(
+        /^[a-z]+$/,
+        'Username can only contain lowercase letters, no spaces, numbers, or special characters.',
+      )
+      .min(3, 'Username must be at least 3 characters long.')
+      .max(32, 'Username must be 32 characters or less.')
+      .trim(),
+    password: z
+      .string({ required_error: 'Password is required.' })
+      .min(1, 'Password is required.')
+      .regex(
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]+$/,
+        'Password must contain at least one uppercase letter, one lowercase letter, and one special character.',
+      )
+      .min(8, 'Password must be at least 8 characters long.')
+      .max(64, 'Password must be 64 characters or less.'),
```

Comment:
This may be a personal thing, but I prefer to check the length in the regular expression and adapt that error message to mention that the length is not correct, so if that error appears and the rest is not fulfilled either you can adjust it and not receive another error later.
But then again, is open for discussion 😄

Reply:
Good point! I hadn't thought of that. I’ll consider combining the length check into the regex. Thanks for the suggestion!